cmd/logcli/main.go

```diff
@@ -358,6 +370,25 @@ func main() {
 			ColoredOutput: rangeQuery.ColoredOutput,
 		}
 
+		switch *timestamp {
```

Comment:
I like this feature, but I have some feedback on the arguments. `nanos` as RFC3339Nano is confusing to me, as the Loki API accepts a nanosecond epoch time as well, which is what I guess in this case would actually be a `unix` timestamp, but I think that will trip people up. I rather be more explicit.

Furthermore, could we add some logic for detecting the timestamp? For example, when [parsing the timestamp in Loki](https://github.com/grafana/loki/blob/21cfa331312c2c15d833b7a7098450c329fcb92e/pkg/loghttp/params.go#L163) we automatically detect the timestamp type. Not sure if we can capture all of these, but I think we could capture a bunch, wdyt?

Reply:
> I like this idea. I'm curious if we could try to do a better job of automatically detecting the timestamp type? How far could we get without introducing a new argument?

This is an output timestamp so it should be provided, not detected. Maybe change in arg name/parameter?

Reply:
> be more explicit

updated in latest commit to use the lowercased time.XXX name instead

Reply:
yeah, if we want to keep this as just an output change, let's change the parameter to indicate that. however, it would be neat if we could accept more formats for the input timestamps as well, in which case the auto-detection would be neat, but maybe outside the scope of this PR? where I see the overlap is we could default the output format to match the input unless overridden?

Reply:
What would you suggest the arg name? For its value should there be support for all the `time.Parse` [formats](https://github.com/golang/go/blob/24b395119b4df7f16915b9f01a6aded647b79bbd/src/time/format.go#L1023-L1031) (as in `--timestamp-layout="xxx"`)? Should the aliases (rfc, stamp, etc) be dropped?

As for input timestamps, I'm sorry I don't follow for maybe I'm not using loki/logcli enough. But when does logcli expect a non standard (non-loki) input timestamps?

If my guess you are referring to the `--from` and `--to` values then I agree it is outside the scope of this PR. (Although the flag name should be more clear since the `--timezone` is already used for output modifier)

Reply:
Yes, I'm referring to `--from` and `--to`, as [logcli only accepts timestamps as RFC3339Nano](https://github.com/grafana/loki/blob/eda3ba865bd1b30ae69b6006e6da31960daac5e6/cmd/logcli/main.go#L615) When I first saw your PR I was hoping it would deal with that side too, as I find it really annoying that I can't use the same timestamps with logcli as I can in the API. Also, if we went that route, it could be neat to default the output to the format of the input, which could be automatically detected. 

If you'd prefer to stick with only addressing output timestamps, we should rename the argument to `--output-timestamp` or `--output-timestamp-format`.

Reply:
Latest commit update only the argument name.

This is my [incomplete attempt](https://github.com/chu121su12/loki/commit/4e6969d3d8c76f579e23d86444a4eee762c04d90) ([relevant tests](https://github.com/chu121su12/loki/blob/4e6969d3d8c76f579e23d86444a4eee762c04d90/pkg/logcli/query/query_test.go#L900-L940)) at detecting from/to format and using it for output formatting. This version would immediately introduces breaking changes due to the man line stating the value should be in "nano without timestamp" but the output formatting (not adjustable) is in `time.RFC3339`.

https://github.com/grafana/loki/blob/e3e1f096ea50cdcea8d64e726827368af14702a8/cmd/logcli/main.go#L62-L64

Let me know if you want to pursue this version instead.